src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -0,0 +1,255 @@
+     *
+     * For unmapped reads, always returns {@link ReadConstants#UNSET_POSITION}
+     *
+     * @return The alignment end (1-based, inclusive) adjusted for clipped bases,
+     *         or {@link ReadConstants#UNSET_POSITION} if the read is unmapped.
+     */
+    int getUnclippedEnd();
+
+    /**
+     * @return The contig that this read's mate is mapped to, or {@code null} if the mate is unmapped
+     * @throws IllegalStateException if the read is not paired (has no mate)
+     */
+    String getMateContig();
+
+    /**
+     * @return The alignment start (1-based, inclusive) of this read's mate, or {@link ReadConstants#UNSET_POSITION}
+     *         if the mate is unmapped.
+     * @throws IllegalStateException if the read is not paired (has no mate)
+     */
+    int getMateStart();
+
+    /**
+     * @return The observed length of the fragment (equivalent to TLEN in SAM), or 0 if unknown.
+     *         Negative if the mate maps to a lower position than the read.
+     */
+    int getFragmentLength();
+
+    /**
+     * @return The mapping quality of this alignment, representing how likely the read maps to this position as
+     *         opposed to other locations. Returns {@link ReadConstants#NO_MAPPING_QUALITY} if there is none.
+     */
+    int getMappingQuality();
+
+    /**
+     * @return The read sequence as ASCII bytes ACGTN=, or an empty byte[] if no sequence is present.
+     */
+    byte[] getBases();
+
+    /**
+     * @return Base qualities as binary phred scores (not ASCII), or an empty byte[] if base qualities are not present.
+     */
+    byte[] getBaseQualities();
+
+    /**
+     * @return Cigar object describing how the read aligns to the reference, or an empty Cigar object if no cigar is present.
+     */
+    Cigar getCigar();
+
+    /**
+     * @return The ID of the read group this read belongs to, or {@code null} for none.
+     */
+    String getReadGroup();
+
+    /**
+     * @return Total number of reads in this read's fragment, or 0 if unknown.
+     */
+    int getNumberOfReadsInFragment();
+
+    /**
+     * @return The read number in sequencing; 1 for first of pair, 2 for second of pair.
+     *         1-based and <= {@link #getNumberOfReadsInFragment}
+     */
+    int getReadNumber();
+
+    /**
+     * @return True if this read is paired (ie., {@link #getNumberOfReadsInFragment} == 2), otherwise false.
+     * @throws org.broadinstitute.hellbender.exceptions.GATKException.MissingReadField if this information is not available
+     */
+    boolean isPaired();
+
+    /**
+     * @return True if this read is paired and the orientation and the distance between reads from the fragment are
+     *         consistent with the sequencing protocol, otherwise false.
+     * @throws org.broadinstitute.hellbender.exceptions.GATKException.MissingReadField if this information is not available
+     */
+    boolean isProperlyPaired();
```

Comment:
it is going to be pretty confusing to have a isPaired and isProperlyPaired operations